networkx/linalg/laplacianmatrix.py

```diff
@@ -51,12 +54,143 @@ def laplacian_matrix(G, nodelist=None, weight="weight"):
+def distance_matrix(G,nodelist=None, weight="weight"):
+    
+    """
+    Notes
+    --------
+    A distance between two nodes is the length of the shortest path between these nodes.
+    Is 0 if is between node with itself and infinity if there is no path joining these two nodes
+    (In case of disconected graph).
+    
+    This is a distance function. Given a graph, it return a pairwise distance matrix
+    between nodes. The graph structure is created through networkx library.
+    
+    
+    Parameters
+    ----------
+    G: Graph (Undirected)
+    
+    Return 
+    ------
+    Pairwise matrix of distance Between Node
+    
+    Raises
+    ------
+    When The graph is disconnected or directed.
+    
+    
+    Example
+    -------
+    A graph of 5 nodes and 5 edges. 
+    
+    >>> G=nx.Graph() # create an empty graph
+    >>> Edges=[(1,2),(2,3),(2,5),(3,4),(4,5)] # Create a list of edges
+    >>> G.add_edges_from(Edges) # Add these edges to the empty graph G.
+    
+
+    >>> print(distance_matrix(G))
+    array([[0, 1, 2, 3, 2],
+       [1, 0, 1, 2, 1],
+       [2, 1, 0, 1, 2],
+       [3, 2, 1, 0, 1],
+       [2, 1, 2, 1, 0]])
+    """
+    if nodelist is None:
+        nodelist = list(G)
+    
+    if nx.is_connected(G):
+        D=list(nx.shortest_path_length(G))
+        D=sorted(D,reverse=False)
+        m=list()
+        for i in range(len(D)):
+            k=sorted(D[i][1].items())
+            for l in range(len(k)):
+                b=k[l][1]
+                m.append(b)
+        D1=np.array(m)
+        D1=D1.reshape(len(D),len(D))
+        return D1
+    else:
+        raise nx.NetworkXError("{} is not connected".format(G))
+
+@not_implemented_for("directed")
+def k_path_Laplacian(G,k, nodelist=None, weight="weight"):
+    
+    
+    """
+    Notes
+    -----
+    k_path_Laplacian is the generalization of standard Laplacian, this one is 1-degree or 
+    local or direct neighbors or distance 1. Unlike k_path_Laplacian where k means the distance 
+    two consider. You put 1 in the Adjacency matrix if the distance between two nodes is k.
+    We call this adjacency matrix k-Path Adjacency matrix.
+    
+    From this Adjacency matrix, you compute the Degree and Laplacian Matrix. You need to know the
+    diameter of your graph (Maximum distance of nodes in a graph.)
+    
+    Parameters
+    ----------
+    k: int
+        Is used to precise distance that you want to use.
+    If k= 2, you put 1 in the matrix if the distance between two nodes is 2. And so on. 
+    
+    Returns
+    -------
+    P: k-Path Adjacency matrix
+    Deg: k Degree matrix
+    Deg-P: k-path Laplacian
+    
+    Example
+    -------
+    
+    >>> k= 2
+    >>> k_path_Laplacian(G,k)[0]
+    
+    array([[0., 0., 1., 0., 1.],
+       [0., 0., 0., 1., 0.],
+       [1., 0., 0., 0., 1.],
+       [0., 1., 0., 0., 0.],
+       [1., 0., 1., 0., 0.]])
+       
+    Raises
+    ------
+    Errors when the graph is not connected or directed.
+       
```

Comment:
I haven't read through this function carefully yet, but you should give a reference for more information on what the k-path laplacian is and why it is useful. That could be as simple as a wikipedia article, or to a paper where it is introduced.  If this idea has been extended to weighted edges, it would be good to include that case as well in this same function.